hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/type/converter/ToConverters.java

```diff
@@ -30,15 +29,18 @@
 import java.util.Map;
 import java.util.Objects;
 
+import static com.hazelcast.sql.impl.type.QueryDataTypeFamily.MAP;
+import static com.hazelcast.sql.impl.type.QueryDataTypeFamily.OBJECT;
+
 public final class ToConverters {
     private static final Map<QueryDataType, ToConverter> CONVERTERS = prepareConverters();
 
     private ToConverters() { }
 
     @Nonnull
     public static ToConverter getToConverter(QueryDataType type) {
-        if (type.getTypeFamily() == QueryDataTypeFamily.OBJECT) {
-            // User-defined types are subject to the same conversion rules as ordinary OBJECT.
+        if (type.getTypeFamily() == OBJECT || type.getTypeFamily() == MAP) {
+            // User-defined types and maps are subject to the same conversion rules as ordinary OBJECT.
```

Comment:
MAP never has `objectTypeName` (at least at present), you might add MAP directly to `CONVERTERS`

Reply:
1. The purpose of `ToConverter`s is to do the reverse of _normalization_: For example, `VARCHAR_CHARACTER.normalize('c') == "c"` and `ToConverters.getToConverter(VARCHAR_CHARACTER).convert("c") == 'c'`. 
2. `Converter.convertToSelf()`, and consequently `QueryDataType.convert()`, are also doing normalization. For example, `Converters.getConverter(Character.class).convertToSelf('c') == "c"` and `VARCHAR_CHARACTER.convert('c') == "c"`.

I think that item 2 is incorrect: `Converter.convertToSelf()` should convert values its own type, not the one expected by SQL, which can already be done via `QueryDataType.normalize()`. That's why I deleted `ToConverter`s by moving their conversion logic to their respective `Converter`s. Now, `Converter.convertToSelf()`, `QueryDataType.convert()` and `QueryDataType.normalize()` work as expected. (d06ba306)